### PR TITLE
Rename terraform to tofu in `GoString` method and docstrings

### LINKED ISSUE #571: Swap 'Terraform' occurrences with 'OpenTofu', and 'terraform' with 'tofu' in the docstrings and `GoString` methods
### OpenTofu Version

```shell
1.6.0-beta1
```

### Use Cases

To ensure clean cut for separation between terraform and tofu.

### Attempted Solutions

-

### Proposal

The following refactoring is proposed:

- In docstring and error messages, substitute the word "Terraform" with the word "OpenTofu".
- In `GoString`/`ToString` methods, substitute the word "terraform"  with the word "tofu", or another "link" according to the package name.
- Exceptions: 
   - Package `internal/registry`.
   - Registry URL. For example, [internal/tofu/context_test.go#L1014](https://github.com/opentofu/opentofu/blob/0930a908227376f0647f759cdfbadfc71f32fad5/internal/tofu/context_test.go#L1014).
   - Links to the issues in the terraform repo. For example, [internal/tofu/context_test.go#L2498](https://github.com/opentofu/opentofu/blob/0930a908227376f0647f759cdfbadfc71f32fad5/internal/tofu/context_apply_test.go#L2498).
   - Docstring text which contains notes about old Tarraform versions. For example, [internal/tofu/context_test.go#L9181](https://github.com/opentofu/opentofu/blob/0930a908227376f0647f759cdfbadfc71f32fad5/internal/tofu/context_apply_test.go#L9181).

#### [internal/tofu/variables.go#L15-L32](https://github.com/opentofu/opentofu/blob/0930a908227376f0647f759cdfbadfc71f32fad5/internal/tofu/variables.go#L15-L32)

```diff
// InputValue represents a raw value for a root module input variable as
-// provided by the external caller into a function like terraform.Context.Plan.
+// provided by the external caller into a function like tofu.Context.Plan.
//
// InputValue should represent as directly as possible what the user set the
// variable to, without any attempt to convert the value to the variable's
// type constraint or substitute the configured default values for variables
-// that wasn't set. Those adjustments will be handled by Terraform Core itself
+// that wasn't set. Those adjustments will be handled by OpenTofu Core itself
// as part of performing the requested operation.
//
-// A Terraform Core caller must provide an InputValue object for each of the
+// A OpenTofu Core caller must provide an InputValue object for each of the
// variables declared in the root module, even if the end user didn't provide
// an explicit value for some of them. See the Value field documentation for
// how to handle that situation.
//
-// Terraform Core also internally uses InputValue to represent the raw value
+// OpenTofu Core also internally uses InputValue to represent the raw value
// provided for a variable in a child module call, following the same
// conventions. However, that's an implementation detail not visible to
// outside callers.
```

#### [internal/tofu/context_apply_test.go#L2522-L2524](https://github.com/opentofu/opentofu/blob/0930a908227376f0647f759cdfbadfc71f32fad5/internal/tofu/context_apply_test.go#L2522-L2524)

```diff
// We'll marshal and unmarshal the plan here, to ensure that we have
// a clean new context as would be created if we separately ran
-// terraform plan -out=tfplan && terraform apply tfplan
+// tofu plan -out=tfplan && tofu apply tfplan
```

#### [internal/tofu/variables.go#L110-L116](https://github.com/opentofu/opentofu/blob/0930a908227376f0647f759cdfbadfc71f32fad5/internal/tofu/variables.go#L110-L116):

```diff
func (v *InputValue) GoString() string {
	if (v.SourceRange != tfdiags.SourceRange{}) {
-		return fmt.Sprintf("&terraform.InputValue{Value: %#v, SourceType: %#v, SourceRange: %#v}", v.Value, v.SourceType, 
+		return fmt.Sprintf("&tofu.InputValue{Value: %#v, SourceType: %#v, SourceRange: %#v}", v.Value, v.SourceType, v.SourceRange)
	} else {
-		return fmt.Sprintf("&terraform.InputValue{Value: %#v, SourceType: %#v}", v.Value, v.SourceType)
+		return fmt.Sprintf("&tofu.InputValue{Value: %#v, SourceType: %#v}", v.Value, v.SourceType)
	}
}
```

### References

_No response_

### PR DESCRIPTION
**Pull request by kislerdm**
Resolves #571 

## Target Release

1.6.0-beta.1

### ENHANCEMENTS

- `terraform` as command is swapped with `tofu` in the `GoString` method and docstrings;
- Terraform is swapped with OpenTofu in docstrings.

### CODE REVIEW

**Comment by kislerdm** [comment]
@cube2222 Hey Kuba! Could you please review/assign reviewers? Thanks!

**Comment by cube2222** [comment]
@kislerdm Hey, thanks for working on this! Looks like some files aren't goimports-formatted.

**Comment by kislerdm** [comment]
@cube2222 indeed - a consequence of my IDE messing with it. I will fix it ASAP. BTW is running the script to check format a part of instructions? Shall we consider pre-commit, or pre-push maybe?

**Comment by cube2222** [comment]
> BTW is running the script to check format a part of instructions? Shall we consider pre-commit, or pre-push maybe?

Good point, I don't believe it is! Feel free to create a separate issue for this.

File: internal/command/jsonchecks/checks.go [hunk 1]
```diff
@@ -70,7 +70,7 @@ func MarshalCheckStates(results *states.CheckResults) []byte {
 // checkResultStatic is the container for the static, configuration-driven
 // idea of "checkable object" -- a resource block with conditions, for example --
 // which ensures that we can always say _something_ about each checkable
-// object in the configuration even if Terraform Core encountered an error
+// object in the configuration even if OpenTofOpenTofu Core encountered an error
```
Comment (cube2222) [inline internal/command/jsonchecks/checks.go:73]: Something went wrong here

File: internal/command/jsonstate/state.go [hunk 2]
```diff
@@ -104,10 +104,10 @@ type Resource struct {
 	// addresses relative to the containing module.
 	DependsOn []string `json:"depends_on,omitempty"`
 
-	// Tainted is true if the resource is tainted in terraform state.
+	// Tainted is true if the resource is tainted in tofutofutofu state.
```
Comment (cube2222) [inline internal/command/jsonstate/state.go:107]: Something went wrong here.

File: internal/command/testdata/login-oauth-server/oauthserver.go [hunk 3]
```diff
@@ -16,9 +16,9 @@ import (
 // Handler is an implementation of net/http.Handler that provides a stub
 // OAuth server implementation with the following endpoints:
 //
-//     /authz  - authorization endpoint
-//     /token  - token endpoint
-//     /revoke - token revocation (logout) endpoint
+//	/authz  - authorization endpoint
```
Comment (cube2222) [inline internal/command/testdata/login-oauth-server/oauthserver.go:19]: I believe this want unintended.

File: internal/command/testdata/login-tfe-server/tfeserver.go [hunk 4]
```diff
@@ -17,8 +17,8 @@ const (
 // Handler is an implementation of net/http.Handler that provides a stub
 // TFE API server implementation with the following endpoints:
 //
-//     /ping            - API existence endpoint
-//     /account/details - current user endpoint
+//	/ping            - API existence endpoint
```
Comment (cube2222) [inline internal/command/testdata/login-tfe-server/tfeserver.go:20]: I believe this was unintended.

File: internal/configs/test_file.go [hunk 5]
```diff
@@ -175,13 +175,13 @@ type TestRunOptions struct {
 	// Mode is the planning mode to run in. One of ['normal', 'refresh-only'].
 	Mode TestMode
 
-	// Refresh is analogous to the -refresh=false Terraform plan option.
+	// Refresh is analogous to the -refresh=false OpenTofOpenTofuu plan option.
```
Comment (cube2222) [inline internal/configs/test_file.go:178]: Something went wrong here.

File: internal/configs/test_file.go [hunk 6]
```diff
@@ -175,13 +175,13 @@ type TestRunOptions struct {
...
 	Refresh bool
 
-	// Replace is analogous to the -refresh=ADDRESS Terraform plan option.
+	// Replace is analogous to the -refresh=ADDRESS OpenTofOpenTofuu plan option.
```
Comment (cube2222) [inline internal/configs/test_file.go:181]: Something went wrong here.

File: internal/getproviders/types.go [hunk 7]
```diff
@@ -291,7 +291,7 @@ type PackageLocation interface {
 // PackageLocalArchive is the location of a provider distribution archive file
 // in the local filesystem. Its value is a local filesystem path using the
 // syntax understood by Go's standard path/filepath package on the operating
-// system where Terraform is running.
+// system where OpenTofuOpenTofu is running.
```
Comment (cube2222) [inline internal/getproviders/types.go:294]: Something went wrong here.

File: internal/legacy/helper/schema/resource.go [hunk 8]
```diff
@@ -127,14 +127,14 @@ type Resource struct {
 	Exists ExistsFunc
 
 	// CustomizeDiff is a custom function for working with the diff that
-	// Terraform has created for this resource - it can be used to customize the
+	// OpenTofuOpenTofu has created for this resource - it can be used to customize the
```
Comment (cube2222) [inline internal/legacy/helper/schema/resource.go:130]: Something went wrong here.

File: internal/command/jsonchecks/checks.go (hunk 1 above)
Comment (kislerdm) [reply internal/command/jsonchecks/checks.go:73]: Thank you for spotting - applied.

File: internal/command/jsonstate/state.go (hunk 2 above)
Comment (kislerdm) [reply internal/command/jsonstate/state.go:107]: Thank you for spotting - applied

File: internal/command/meta_backend.go [hunk 9]
```diff
@@ -73,8 +73,8 @@ type BackendWithRemoteTerraformVersion interface {
 
 // Backend initializes and returns the backend for this CLI session.
 //
-// The backend is used to perform the actual Terraform operations. This
-// abstraction enables easily sliding in new Terraform behavior such as
+// The backend is used to perform the actual OpenTofutofutofu operations. This
```
[the remark this answers is not in the document]
Comment (kislerdm) [reply internal/command/meta_backend.go:76]: Thank you for spotting - applied

File: internal/tofu/variables.go [hunk 10]
```diff
@@ -13,20 +13,20 @@ import (
...
 
 // InputValue represents a raw value for a root module input variable as
-// provided by the external caller into a function like terraform.Context.Plan.
+// provided by the external caller into a function like tofu.Context.Plan.
 //
 // InputValue should represent as directly as possible what the user set the
 // variable to, without any attempt to convert the value to the variable's
 // type constraint or substitute the configured default values for variables
-// that wasn't set. Those adjustments will be handled by Terraform Core itself
+// that wasn't set. Those adjustments will be handled by OpenOpenTofu Core itself
```
Comment (cube2222) [inline internal/tofu/variables.go:21]: Something went wrong here.

File: internal/configs/test_file.go (hunk 6 above)
Comment (kislerdm) [reply internal/configs/test_file.go]: Thank you for spotting - applied

File: internal/getproviders/types.go (hunk 7 above)
Comment (kislerdm) [reply internal/getproviders/types.go]: Thank you for spotting - applied

File: internal/legacy/helper/schema/resource.go (hunk 8 above)
Comment (kislerdm) [reply internal/legacy/helper/schema/resource.go]: Thank you for spotting - applied

File: internal/command/testdata/login-oauth-server/oauthserver.go (hunk 3 above)
Comment (kislerdm) [reply internal/command/testdata/login-oauth-server/oauthserver.go:19]: It's the result of running `gofmt` and `goimport` in my local env 🤔

File: internal/tofu/variables.go (hunk 10 above)
Comment (kislerdm) [reply internal/tofu/variables.go]: Thank you for spotting - applied

File: internal/command/testdata/login-oauth-server/oauthserver.go (hunk 3 above)
Comment (cube2222) [reply internal/command/testdata/login-oauth-server/oauthserver.go:19]: Ok, let's leave it then.

**Review by cube2222 — APPROVED** [review]

File: internal/tofu/eval_variable.go [hunk 11]
```diff
@@ -228,7 +228,7 @@ func evalVariableValidations(addr addrs.AbsInputVariableInstance, config *config
 		diags = diags.Append(&hcl.Diagnostic{
 			Severity: hcl.DiagError,
 			Summary:  "No final value for variable",
-			Detail:   fmt.Sprintf("OpenTofu doesn't have a final value for %s during validation. This is a bug in Terraform; please report it!", addr),
+			Detail:   fmt.Sprintf("OpenTofu doesn't have a final value for %s during validation. This is a bug in OpenTofu; please report it!", addr),
```
Comment (marcinwyszynski) [inline internal/tofu/eval_variable.go:231]: Oops, good catch.

Comment (kislerdm) [reply internal/tofu/eval_variable.go:231]: Thanks! `grep` has alway been enabling for "unexpected" discoveries 😄

**Pull request merged**